### PR TITLE
codingstyle: Added Python style guide.

### PR DESCRIPTION
**Pull request by pepol**
Signed-off-by: Peter Polacik <pepol@imterra.com>

### CODE REVIEW

File: python_guide.md [hunk 1]
```diff
@@ -0,0 +1,306 @@
+# Coding Guidelines for Python
+
+Python is the main scripting language used at Imterra. This style guide is a
+list of *dos* and *don'ts* for Python programs.
+
+This style guide **mostly** follows
+[PEP-8][https://www.python.org/dev/peps/pep-0008/], but differs from it in few
```
Comment (pepol) [inline python_guide.md]: This should be [...](...), will fix this.

File: python_guide.md [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+### Comments
+
+Comments must be kept up-to-date with any code changes!
+
+Comments should be complete sentences. If a comment is a phrase or a sentence,
+its first word should be capitalized, unless it is an identifier that begins
+with a lower case letter (**never alter the case of identifiers**). Comments
+should end with period (`.`), just like normal sentences.
+
+All comments must be in English.
```
Comment (JakubNvk) [inline python_guide.md]: @pepol wouldnt it be prettier to have "must be written in"?

Comment (pepol) [inline python_guide.md]: Yeah I guess so...

File: python_guide.md [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+date or event.
+
+### Naming conventions
+
+**Names to avoid:**
+
+ - single character names, except for counters or iterators.
+ - dashes (`-`) in any package/module name.
+ - double leading and trailing underscore (`__`) names (reserved by Python).
+
```
Comment (pepol) [inline python_guide.md]: I should add some text here to separate those two lists visually. Probably something like: "Use this list as a reference for identifier style:" What's your opinion @JakubNvk?

File: python_guide.md [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+   you really mean `if x is not None`.
+ - Use `is not` operator rather than `not ... is`. The former expression is
+   more readable and thus preferred.
+ - Always use a `def` statement instead of an assignment statement that binds
+   a `lambda` expression directly to an identifier.
+ - When raising an exception, use `raise MyException('message')` instead of
+   `raise MyException, 'message'`, since the latter form is not legal in
+   Python 3.
+ - When catching exceptions, don't use catch-all `except` clauses.
+   **Exception (pun intended):** Unittests.
```
Comment (JakubNvk) [inline python_guide.md]: +1

File: python_guide.md (hunk 3 above)
Comment (JakubNvk) [inline python_guide.md]: @pepol yes, that'd be better.

**Comment by pepol** [comment]
@JakubNvk ping

**Comment by JakubNvk** [comment]
@pepol LGTM!

**Pull request merged**